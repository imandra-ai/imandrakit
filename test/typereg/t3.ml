open Util_

(* test that typereg.name works to disambiguate types *)

module A = struct
  type t = int [@@deriving typereg] [@@typereg.name "A.t"]
end

module B = struct
  type t = bool [@@deriving typereg] [@@typereg.name "B.t"]
end

type u = { x: float } [@@unboxed] [@@deriving typereg]

let () =
  print_endline "t3:";
  dump ()
