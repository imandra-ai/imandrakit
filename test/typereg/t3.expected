t3:
{ path = "Dune__exe__T3"; name = "A.t"; params = [];
  decl = (Alias (Cstor ("int", []))) }
{ path = "Dune__exe__T3"; name = "B.t"; params = [];
  decl = (Alias (Cstor ("bool", []))) }
